### PR TITLE
Fix issues with running volume tests as non-root.

### PR DESCRIPTION
**Pull request by cpuguy83**
- Volume store created dir with wrong permissions
- Local volume driver hardcoded uid/gid 0

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM :tada: 
```bash
$ go test -v ./volume/store
# […]
ok      github.com/docker/docker/volume/store   0.006s
```

File: volume/store/store.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func New(rootPath string, drivers *drivers.Store) (*VolumeStore, error) {
 	if rootPath != "" {
 		// initialize metadata store
 		volPath := filepath.Join(rootPath, volumeDataDir)
-		if err := os.MkdirAll(volPath, 750); err != nil {
+		if err := os.MkdirAll(volPath, 0750); err != nil {
```
Comment (thaJeztah) [inline volume/store/store.go]: oh! what were the effective permissions due to this?

**Review by thaJeztah — APPROVED** [review]
LGTM 😍

File: volume/store/store.go (hunk 1 above)
Comment (cpuguy83) [reply volume/store/store.go]: `drwx------`

**Pull request merged**